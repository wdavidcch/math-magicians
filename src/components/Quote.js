import React from 'react';

export default function Quote() {
  return (
    <div>
      <h3>
        Mathematics is not about numbers, equations, computations, or
        algorithms: it is about understanding. -William Paul Thurston
      </h3>
    </div>
  );
}
