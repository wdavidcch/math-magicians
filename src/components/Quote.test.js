import { render, screen } from '@testing-library/react';
import Quote from './Quote';
import '@testing-library/jest-dom';

describe('Tests for quotes page', () => {
  test('renders correctly snapshot', () => {
    const tree = render(<Quote />);
    expect(tree).toMatchSnapshot();
  });
  test('Author`s name is on the page', () => {
    render(<Quote />);
    expect(screen.getByText(/-William Paul Thurston/)).toBeInTheDocument();
  });
  test('Mathematics is on the page', () => {
    render(<Quote />);
    expect(screen.queryByText(/Mathematics/)).toBeInTheDocument();
  });
});
