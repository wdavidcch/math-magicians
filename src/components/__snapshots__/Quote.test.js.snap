// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for quotes page renders correctly snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="display: flex; flex-direction: column; align-items: center; justify-content: center; min-height: 60vh;"
      >
        <h3
          style="text-align: center; font-size: 2rem; width: 70%;"
        >
          Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
        </h3>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="display: flex; flex-direction: column; align-items: center; justify-content: center; min-height: 60vh;"
    >
      <h3
        style="text-align: center; font-size: 2rem; width: 70%;"
      >
        Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. -William Paul Thurston
      </h3>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
